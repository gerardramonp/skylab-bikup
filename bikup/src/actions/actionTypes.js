export default {
    LOAD_BIKES: 'LOAD_BIKES',
    LOAD_BIKE_BY_ID: 'LOAD_BIKE_BY_ID'
};
