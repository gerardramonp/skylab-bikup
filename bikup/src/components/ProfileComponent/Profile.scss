$light-bg-color: white;
$light-highlight-color: #eeeeee;
$main-color: #0c9;
$dark-highlight-color: #393e46;
$strava-button-bg-color: #fc4c02;
$strava-button-color: whitesmoke;
$standard-button-color: black;
$header-bg-color: #222831;

$light-login-bg: rgba(238, 238, 238, 0.6);

$card-shadow: 2px 2px 5px 0px rgba(0, 0, 0, 0.514);

.profile {
	display: flex;
	flex-direction: column;
	padding: 12px;
}

.profile__container {
	display: flex;
	flex-direction: column;
	align-items: center;
}

.profile__picture {
	display: flex;
	flex-direction: row;
	justify-content: center;
}

.picture__img {
	border-radius: 50%;
	height: 150px;
}
