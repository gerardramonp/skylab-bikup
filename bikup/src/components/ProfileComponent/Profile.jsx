import React, { useState, useEffect } from 'react';
import { useHistory } from 'react-router-dom';

import { isUserAuthWithToken } from '../../actions/authActions';

import authStore from '../../stores/authStore';

import './Profile.scss';

let isUserAuth = null;
let userCheck = false;
let userInfo = null;

function Profile() {
	let [userInfo, setUserInfo] = useState();

	const history = useHistory();

	async function handleAuthorization() {
		await checkIfUserIsAuth();
	}

	async function checkIfUserIsAuth() {
		await isUserAuthWithToken();
		isUserAuth = authStore.isUserAuth();
		if (!isUserAuth) {
			history.replace('/login');
		} else {
			userCheck = true;
			userInfo = JSON.parse(sessionStorage.authUser);
		}
	}

	useEffect(() => {
		authStore.addChangeListener(onAuthChange);
		handleAuthorization();
	}, [isUserAuth]);

	function onAuthChange() {
		isUserAuth = authStore.isUserAuth();
		isUserAuth &&
			setUserInfo((userInfo = JSON.parse(sessionStorage.authUser)));
	}

	return (
		(userInfo && (
			<>
				<div className='general-container'>
					<div className='profile'>
						<div className='profile__container'>
							<div className='profile__picture'>
								<img
									className='picture__img'
									src={userInfo.profilePicture}
									alt='User profile picture'
								/>
							</div>
							<div className='profile__main'>
								<h3>{userInfo.username} Profile</h3>
								<div className='main__info'>
									<p>
										Connected with strava:{' '}
										{userInfo.stravaAccessToken
											? 'yes'
											: 'no'}
									</p>
								</div>
							</div>
							<div className='profile__edit'></div>
						</div>
					</div>
				</div>
			</>
		)) || <p>loading...</p>
	);
}

export default Profile;
