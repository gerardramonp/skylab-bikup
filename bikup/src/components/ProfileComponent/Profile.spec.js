import React from 'react';
import renderer from 'react-test-renderer';
import Profile from './Profile';
import { BrowserRouter as Router } from 'react-router-dom';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';
import dispatcher from '../../dispatcher';
import actionTypes from '../../actions/actionTypes';

Enzyme.configure({ adapter: new Adapter() });

describe('Profile Component', () => {
	let tree = null;
	let compo = null;
	let mountedCompo = null;

	beforeEach(() => {
		tree = renderer.create(
			<Router>
				<Profile />
			</Router>
		);
	});

	it('should match', () => {
		dispatcher.dispatch({
			type: actionTypes.LOGIN_USER_STRAVA,
			data: true,
		});
		expect(tree.toJSON()).toMatchSnapshot();
	});
});
