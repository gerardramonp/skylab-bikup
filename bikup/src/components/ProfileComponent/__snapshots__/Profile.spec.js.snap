// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Component should match 1`] = `
<p>
  loading...
</p>
`;
